src/nvim/macros.h

```diff
@@ -92,29 +92,18 @@
 
 #define REPLACE_NORMAL(s) (((s) & REPLACE_FLAG) && !((s) & VREPLACE_FLAG))
 
-#define UTF_COMPOSINGLIKE(p1, p2)  utf_composinglike((p1), (p2))
 
 // MB_PTR_ADV(): advance a pointer to the next character, taking care of
-// multi-byte characters if needed.
-// MB_PTR_BACK(): backup a pointer to the previous character, taking care of
-// multi-byte characters if needed.
-// MB_COPY_CHAR(f, t): copy one char from "f" to "t" and advance the pointers.
-// PTR2CHAR(): get character from pointer.
-
-// Advance multi-byte pointer, skip over composing chars.
+// multi-byte characters if needed. Skip over composing chars.
 #define MB_PTR_ADV(p)      (p += utfc_ptr2len((char_u *)p))
+
 // Advance multi-byte pointer, do not skip over composing chars.
 #define MB_CPTR_ADV(p)     (p += utf_ptr2len(p))
```

Comment:
Side note: `MB_CPTR_ADV` is only used 3 times, but the other one `MB_PTR_ADV` is used quite a bit in `regex.c` and `spell.c`. However, I do feel like both of them should not really be macros, but maybe `inline` functions at the very least since they deal with raw memory.